x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/SearchableSnapshotIndexInput.java

```diff
@@ -108,19 +197,24 @@ protected void seekInternal(long pos) throws IOException {
         } else if (pos < 0L) {
             throw new IOException("Seeking to negative position [" + pos + "] for " + toString());
         }
-        this.position = offset + pos;
+        if (position != offset + pos) {
+            position = offset + pos;
+            IOUtils.close(streamForSequentialReads);
+            streamForSequentialReads = null;
+        }
     }
 
     @Override
     public BufferedIndexInput clone() {
-        return new SearchableSnapshotIndexInput("clone(" + this + ")", blobContainer, fileInfo, position, offset, length);
+        return new SearchableSnapshotIndexInput("clone(" + this + ")", blobContainer, fileInfo, position, offset, length,
+            NO_SEQUENTIAL_READ_OPTIMIZATION, getBufferSize());
```

Comment:
Maybe add a small word on why we can't read optimized for clones?

Reply:
Comments added in 2fee6b851ac4482dd2ec43c46c87debdbeb71113.